markdown/plext/using_plpgsql.html.md.erb

```diff
@@ -19,143 +19,278 @@ software distributed under the License is distributed on an
+-   Extra round trips between client and server are eliminated
+-   Intermediate, and perhaps unneeded, results do not have to be marshaled or transferred between the server and client
+-   You avoid multiple rounds of query parsing
+ 
 
-With PL/pgSQL, you can group a block of computation and a series of queries inside the database server, thus having the power of a procedural language and the ease of use of SQL, but with considerable savings of client/server communication overhead.
+## <a id="plpgsql_structure"></a>PL/pgSQL Function Syntax
 
--   Extra round trips between client and server are eliminated
--   Intermediate results that the client does not need do not have to be marshaled or transferred between server and client
--   Multiple rounds of query parsing can be avoided
+PL/pgSQL is a block-structured language. The complete text of a function definition must be a block, which is defined as:
 
-This can result in a considerable performance increase as compared to an application that does not use stored functions.
+``` sql
+[ <label> ]
+[ DECLARE
+    declarations ]
+BEGIN
+    statements
+END [ label ];
+```
 
-PL/pgSQL supports all the data types, operators, and functions of SQL.
+Each declaration and each statement within a block is terminated by a semicolon. A block that appears within another block must have a semicolon after `END`, as shown above; however the final `END` that concludes a function body does not require a semicolon.
+
+You can specify all key words and identifiers in mixed upper and lower case. Identifiers are implicitly converted to lowercase unless double-quoted.
+
+PL/pgSQL supports two types of comments. A double dash (`--`) starts a comment that extends to the end of the line. A `/*` starts a block comment that extends to the next occurrence of `*/`. Block comments cannot be nested, but you can enclose double dash comments into a block comment and a double dash can hide the block comment delimiters `/*` and `*/`.
+
+This example PL/pgSQL function adds thirteen to an integer:
+
+``` sql
+=> CREATE FUNCTION add_thirteen(i integer) RETURNS integer AS 
+   $$
+   DECLARE
+       incvalue integer := 13;
+   BEGIN
+       -- add thirteen to i
+       RETURN i + incvalue;
+   END;
+   $$ LANGUAGE plpgsql;
+=> SELECT add_thirteen( 11 );
+    increment 
+   -----------
+           24
+   (1 row)
+```
 
-**Note:**  PL/pgSQL is automatically installed and registered in all HAWQ databases.
+**Note**: Do not to confuse the use of `BEGIN/END` for grouping statements in PL/pgSQL with the database commands for transaction control. PL/pgSQL's BEGIN/END are only for statement grouping; they do not start or end a transaction. 
```

Comment:
i don't believe you can use begin/end in a plpgsql block.  i think the whole function is run in a transaction.